### ISSUE TITLE
Value not displaying in Safari inside v-for

### ISSUE THREAD

**Issue by gborcherds** [charliekassel/vuejs-datepicker#167]
This is specific to safari. I'm passing in the data through a loop as in the code below in a table. It works in other browsers, but not safari. I'm on the latest Vue version, just wondering if you might know why it's doing this.

```javascript
<tr v-for="(transfer, index ) in shared.transfers" :key="transfer.id">
                                <td>
                                    <datepicker format="dd MMM yyyy" wrapper-class="d-block" input-class="form-control" :name="'transfer_start_date_'+transfer.id" :id="'transfer_start_date_'+transfer.id" :value="transfer.start_date"></datepicker>
                                </td>
                                <td>
                                    <datepicker format="dd MMM yyyy" wrapper-class="d-block" input-class="form-control" :name="'transfer_end_date_'+transfer.id" :id="'transfer_end_date_'+transfer.id" v-model="transfer.end_date"></datepicker>
                                </td>
                                <td>
                                    <input class="form-control" v-model="transfer.companion" type="text">
                                </td>
                                <td>
                                    <input class="form-control" v-model="transfer.area" type="text">
                                </td>
                                <td>
                                    <button class="btn btn-sm btn-danger" @click="removeTransfer(index)"><i class="fa fa-trash-o"></i></button>
                                </td>
                            </tr>
```

**Comment by charliekassel** [comment]
Ah Safari's new Date constructor with a string is a bit finicky, I'll take a look it shortly

**Comment by charliekassel** [comment]
@gborcherds what is transfer.start_date and transfer.end_date?
Date Object (preferred) or String? If String how is it formatted?

**Comment by gborcherds** [comment]
Y-m-d H:i:s as a string. I could pass a date object though. I'd just need to loop through the array once and conver them myself.

Thank you,
Gareth Borcherds
On Tue, Jun 13, 2017 at 12:27 AM, charliekassel <notifications@github.com> wrote:
@gborcherds [https://github.com/gborcherds] what is transfer.start_date and transfer.end_date?
Date Object (preferred) or String? If String how is it formatted?

**Comment by prine** [comment]
I had the same issue. I solved the problem with using the https://momentjs.com/ Library to parse the date string and reformat it. Because format() is returning the correct date format which is also working in Safari.

`moment(start_date).format();` 

Kind of stupid, but it does its job :).

**Comment by gborcherds** [comment]
Yeah, that's what I ended up doing as well. I didn't want to have to loop through my array and modify each date just cause it's a loop, but that did work.

**Comment by larsemil** [comment]
I have the same issue. Makes v-model kind of hard to work with.

**Comment by tiagomatosweb** [comment]
This issue seems to persist in Safari.
I noticed that creating new Date('YYYY-MM-DD HH:MM:SS') format throws error, while in Chrome does not.

So, I did the said hack, used moment() to reformat the date, like

```
new Date(moment(myDate).format())
```

It because moment will convert the date into ISO 8601 format which is YYYY-MM-DDTHH:MM:SS (T in the middle), then Safari can recognise.

Agree stupid but...

**Comment by 17agustin** [comment]
2023 and this issue is still a fact. I changed all my `new Date()` to moment js and it worked right away
